src/compas/datastructures/mesh/_mesh.py

```diff
@@ -644,17 +645,17 @@ def from_vertices_and_faces(cls, vertices, faces):
         """
         mesh = cls()
 
-        if type(vertices) == list:
+        if isinstance(vertices, collections.MutableSequence):
             for x, y, z in iter(vertices):
                 mesh.add_vertex(x=x, y=y, z=z)
-        elif type(vertices) == dict:
+        elif isinstance(vertices, collections.Mapping):
             for key, xyz in vertices.items():
                 mesh.add_vertex(key = key, attr_dict = {i: j for i, j in zip(['x', 'y', 'z'], xyz)})
         
-        if type(faces) == list:
+        if isinstance(faces, collections.MutableSequence):
```

Comment:
can it not just be `collections.Sequence`?
doesn't have to be a list, can also be a tuple...